modules/lib.nix

```diff
@@ -15,5 +19,14 @@ in {
       inherit (self'.packages) cosmwasm-check;
     };
   };
-  flake.lib = lib std;
+  flake.lib = system:
+    withSystem system ({
+      pkgs,
+      self',
+      ...
+    }:
+      lib std {
+        inherit pkgs;
+        inherit (self'.packages) cosmwasm-check;
+      });
```

Comment:
I think just passing a `system` is a cleaner interface, and allows us to hide some of the specific things we need to do in the `lib`

Reply:
yes. but weird lib depends on system. usually specific builders inside lib depend on system or of lib is in pkgs, but no dep in flake output directly. but sure, system is better.